addons/product/models/product_template.py

```diff
@@ -357,6 +357,7 @@ def create(self, vals_list):
             templates._create_variant_ids()
 
         # This is needed to set given values to first variant after creation
+        self.flush()
```

Comment:
We wonder what's the use of this. Could you specify? Maybe with a comment?

Reply:
To avoid a cache miss. When trying to update the variant values according to the template values, the newly created records should be flushed.

Don't you think that the commit message is not sufficient ? :)

In fact, maybe we could add to the guidelines the fact that all the calls to ``flush`` should be documented in that case.

What do you think ? 

cc @jem-odoo @tde-banana-odoo

Reply:
Well, even with this comment I still don't get it.

> the newly created records should be flushed.

`creates` are not delayed. So if you need a a template id for your variants, it will be there.

Unless you require a specific stored field for the template that has just been created, this call to flush shouldnt be needed. And if this is the case, then should precise the field in the call to `flush` so this is explicit. For instance 
```
self.flush(['amount_total'])
```
What is the use case leading to the issue/traceback?

Reply:
Hi. 

After a small investigation, it seems you're right ! ;)

This commit https://github.com/odoo/odoo/pull/36458/commits/bb505defe4abfb0e051493b47952c9b066a7252c fix both issues. 

Thanks.